testcases/commands/logrotate/logrotate_tests.sh

```diff
@@ -40,342 +40,148 @@
 #                Dec 24 2002 - Added   - Test #2 - Test to run logrotate as a
 #                                        cron job.
 #                Feb 28 2003 - Fixed   - Modified testcase to use functions.
+#                Mar 04 2024 - Fixed   - Rewrite according to the new guideline.
 #
-# Function: 	chk_ifexists
-#
-# Description:  - Check if command required for this test exits.
-#
-# Input:        - $1 - calling test case.
-#               - $2 - command that needs to be checked.
-#
-# Return:		- zero on success.
-# 				- non-zero on failure.
-chk_ifexists()
-{
-	RC=0
-
-	which $2 > $LTPTMP/tst_logrotate.err 2>&1 || RC=$?
-	if [ $RC -ne 0 ]
-	then
-		tst_brkm TBROK NULL "$1: command $2 not found."
-	fi
-	return $RC
-}
 
+TST_NEEDS_CMDS="logrotate egrep file"
+TST_TESTFUNC=test
+TST_NEEDS_TMPDIR=1
+TST_CNT=2
+TST_SETUP=setup
+TST_CLEANUP=cleanup
 
-# Function: init
-#
-# Description:  - Check if command required for this test exits.
-#               - Create temporary directories required for this test.
-#               - Initialize global variables.
-#
-# Return:		- zero on success.
-# 				- non-zero on failure.
-init()
-{
-	# Initialize global variables.
-	export RC=0
-	export TST_TOTAL=2
-	export TCID="logrotate"
-	export TST_COUNT=0
-
-	# Inititalize cleanup function.
-	trap "cleanup" 0
-
-	# create the temporary directory used by this testcase
-	if [ -z $TMP ]
-	then
-		LTPTMP=/tmp/tst_logrotate.$$
-	else
-		LTPTMP=$TMP/tst_logrotate.$$
-	fi
-
-	mkdir -p $LTPTMP > /dev/null 2>&1 || RC=$?
-	if [ $RC -ne 0 ]
-	then
-		 tst_brkm TBROK "INIT: Unable to create temporary directory"
-		 return $RC
-	fi
-
-	# check if commands tst_*, logrotate, awk and file exists.
-	chk_ifexists INIT tst_resm  || return $RC
-	chk_ifexists INIT logrotate || return $RC
-	chk_ifexists INIT awk       || return $RC
-	chk_ifexists INIT file      || return $RC
-
-	return $RC
+setup(){
+	export LOGROTATE=`which logrotate`
```

Comment:
This is used just on single place, no need a setup function. Please use it directly on that place.
Also use `command -v` instead of `which`.